db_stress_tool/db_stress_tool.cc

```diff
@@ -45,6 +45,10 @@ int db_stress_tool(int argc, char** argv) {
   SanitizeDoubleParam(&FLAGS_memtable_prefix_bloom_size_ratio);
   SanitizeDoubleParam(&FLAGS_max_bytes_for_level_multiplier);
 
+  if (FLAGS_use_direct_reads || FLAGS_use_direct_io_for_flush_and_compaction) {
+    test::SetupSyncPointsToMockDirectIO();
```

Comment:
Can we do this conditionally through a separate command line argument set by db_crashtest.py? Otherwise, it precludes the ability to run crash_test with end-to-end direct IO.

Reply:
added --mock_direct_io